src/shared/utils/utils.ts

```diff
@@ -24,6 +22,13 @@ import {
   isWrappedV4Document,
 } from "./guard";
 
+const VersionGuard = {
+  2: isWrappedV2Document,
+  3: isWrappedV3Document,
+  4: isWrappedV4Document,
+} as const satisfies Record<number, (document: unknown) => boolean>;
+type OAVersion = keyof typeof VersionGuard;
```

Comment:
There's another `Version` type in `4.0/diagnose.ts`.

```typescript
type Version = "2.0" | "3.0" | "4.0";
```

Should we unify by utilising it instead of using `number`?

```suggestion
const VersionGuard = {
  "2.0": isWrappedV2Document,
  "3.0": isWrappedV3Document,
  "4.0": isWrappedV4Document,
} as const satisfies Record<Version, (document: unknown) => boolean>;
// type OAVersion = keyof typeof VersionGuard; // Also omit this line in favour of `Version`
```